### PR TITLE
Minor fix: Post OpsGenie alert close requests to /v1/json/alert/close endpoint

### PR DESCRIPTION
**Pull request by amorken**
Closing OpsGenie alerts requires that the close request is POSTed to an endpoint ending with /close but otherwise identical to the alert creation endpoint. Update notifier.go to reflect this and add a test.

Reference: https://www.opsgenie.com/docs/web-api/alert-api#closeAlertRequest ("Sample Request")

### CODE REVIEW

**Comment by amorken** [comment]
I don't actually _know_ golang, but I've tried to mimic the code and style in the other tests. Don't be afraid of treating me like I don't know what I'm doing, as I don't, but at least the test passes - and fails without the fix. ;)

**Comment by amorken** [comment]
Hi @juliusv, I've adopted @jbarratt's approach to testing the URL in the HTTP handler function as you suggested. =)

File: manager/notifier.go [hunk 1]
```diff
@@ -668,7 +671,7 @@ func (n *notifier) sendOpsGenieNotification(op notificationOp, config *pb.OpsGen
 		Timeout: timeout,
 	}
 	resp, err := client.Post(
-		*opsgenieAPIURL,
+		postUrl,
```
Comment (juliusv) [inline manager/notifier.go]: `postUrl` -> `postURL` please :) (https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

Sorry, didn't notice this earlier.

File: manager/notifier.go [hunk 2]
```diff
@@ -652,6 +654,7 @@ func (n *notifier) sendOpsGenieNotification(op notificationOp, config *pb.OpsGen
 			return err
 		}
 	case notificationOpResolve:
+		postUrl = *opsgenieAPIURL + "/close"
 		msg := &opsGenieMessageClose{
 			ApiKey: config.GetApiKey(),
```
Comment (juliusv) [inline manager/notifier.go]: Actually this was that way before, but `ApiKey` should also be renamed to `APIKey`.

File: manager/notifier_test.go [hunk 3]
```diff
@@ -356,6 +357,46 @@ func TestSendOpsGenieNotification(t *testing.T) {
...
+		if err != nil {
+			t.Errorf("error reading webhook notification: %s", err)
+		}
+		if !strings.HasSuffix(r.URL.String(), "/close") {
+			t.Errorf("OpsGenie close notifications must be POSTed to /close endpoint, was posted to %s", r.URL.String())
+		}
+	}))
+	defer ts.Close()
+	opsgenieAPIURL = &ts.URL
+	apikey := "AAAB"
```
Comment (juliusv) [inline manager/notifier_test.go]: `apikey` -> `apiKey`

**Comment by juliusv** [comment]
:+1: otherwise, thanks :)

**Comment by amorken** [comment]
Thanks, @juliusv, I've added some style fixes. In addition to the ones you pointed out, vim-go had some strong opinions (AKA a write hook =) on import ordering and alignment, so I included those while we were at it. =)

**Comment by juliusv** [comment]
@amorken Ah great, thanks! Yeah, I didn't even notice that it wasn't `go fmt`-ed before :) Merging...

**Comment by jbarratt** [comment]
Ha! I was slammed this week and just started working on troubleshooting. Glad I took a look at this before I picked it back up. :heart:

**Pull request merged**